### PR TITLE
doc(multi-staging): fix typos

### PR DESCRIPTION
**Pull request by robstoll**

### CODE REVIEW

**Comment by robstoll** [comment]
The build fails due to dotty.communitybuild.CommunityBuildTest - most likely not because of my changes

File: docs/docs/reference/metaprogramming/staging.md [hunk 1]
```diff
@@ -106,7 +106,7 @@ f.apply(Array(1, 2, 3)) // Returns 6
 ```
 
 Note that if we need to run the main (in an object called `Test`) after
-compilation we need make available the compiler to the runtime:
+compilation we need to make available the compilation to the runtime:
```
Comment (smarter) [inline docs/docs/reference/metaprogramming/staging.md]: The original is actually correct, it's referring to using `-with-compiler` to put the compiler on the classpath.

Comment (robstoll) [reply docs/docs/reference/metaprogramming/staging.md]: Ah... now I see the intention behind it. I'll try to re-formulate it. Some additional questions first though. The current text makes it sound like using `-with-compiler` is only necessary when the code is run via main. The example implies its also necessary for compilation. What's correct? Are there cases where one can compile at runtime but does not require `-with-compiler`?

Comment (smarter) [reply docs/docs/reference/metaprogramming/staging.md]: You need the compiler on the classpath at both compile-time and run-time, so if you're using `dotc` manually you need to pass `-with-compiler`, but actually I think this is not very useful information, 99% of people should just add dotty-staging to their sbt build and not worry about the compiler flags. I4d say the sbt instructions should be at the top of the page and the dotc/dotr stuff removed.

Comment (robstoll) [reply docs/docs/reference/metaprogramming/staging.md]: Good, that's what I thought as well. I'll update the PR later on

**Pull request merged**